initfiles/componentfiles/configschema/xsd/vaults.xsd

```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+################################################################################
+#    HPCC SYSTEMS software Copyright (C) 2022 HPCC Systems®.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+################################################################################
+-->
+
+<xs:schema
+        xmlns:xs="http://www.w3.org/2001/XMLSchema" elementFormDefault="qualified" attributeFormDefault="unqualified"
+        xmlns:hpcc="someuri">
+    <xs:include schemaLocation="types.xsd"/>
+    <hpcc:insert hpcc:schemaPath="/Environment/Software">
+        <xs:element name="vaults" hpcc:displayName="vaults section" hpcc:itemType="vaults" maxOccurs="1" use="required" hpcc:class="component">
+            <xs:complexType>
+                <xs:sequence>
+                    <xs:element name="git" hpcc:displayName="Vault Git Group" hpcc:class="elementSet" maxOccurs="1" use="required">
```

Comment:
Were you able to test w/o the "required" value to make sure that the empty elements weren't added?

Reply:
I attempted to remove the use=required and there was no change. Then I attempted to remove the maxOccurs=1. Still no change. Only when I completely removed the xsd did I no longer get the duplicate <ecl \> entries.  Should I just remove the xsd and keep it as the simple template only in the PR?

Reply:
So for the hpcc:class elementSet, I should have been using minOccurs=0 so it wouldn't force an empty populated element on envmod's first pass through (before it processes the template).  Also removing the use=required from the individual attributes, as that was also causing issues with populating empty elements. With both of those changes its now working how we expect.